### ISSUE TITLE
Mouse movement is slow when using not the default Motion

### ISSUE THREAD

**Issue by seggcsont** [TestFX/TestFX#506]
<pre>
dependency "org.testfx:testfx-core:4.0.8-alpha"
dependency "org.testfx:testfx-junit:4.0.8-alpha"
</pre>

We wanted to use Motion.HORIZONTAL_FIRST for navigating in a context menu, but it is really slow.

During the debugging we realized that moving the mouse by ~300 pixels it takes more than 13000 steps.

We think there is some calculation issue around:
org.testfx.robot.impl.MoveRobotImpl#moveMouseStepwiseBetween
Especially here:
org/testfx/robot/impl/MoveRobotImpl.java:132

To reproduce this just use moveTo(Pointquery, Motion) with Motion.HORIZONTAL_FIRST or Motion.VERTICAL_FIRST

**Comment by brcolow** [comment]
I am sure the problem still occurs on the latest version, 4.0.10-alpha, but it would be nice to confirm anyways. I assume the equivalent line on latest master is line 118 of `MoveRobotImpl.java`. What makes you think it is that line? Do you have a suggestion for fixing it?

**Comment by brcolow** [comment]
I think I found the problem. Working on the fix in #510 - thanks for the report.

**Comment by seggcsont** [comment]
Hi,

I have checked you changes and also implemented our workaround.

The root cause is that percentHorizontal and percentVertical actually not a percent.

As far as I understand this part of the code it calculates how many steps we should use for vertical and horizontal movement.

Like we have two point:
p1 = new Point2D(100,100)
p2 = new Point2D(400,200)
it means the interpolate point will be new Point2D(400, 100).
So the source.distance will be 300 and the target.distance will be 100.
We would like to use 3/4 of the steps for horizintal and 1/4 for vertical movement.
Currently we calculate this percent by source.distance / target,distance.

This is not correct, we should use source.distance / (source.distance + target.distance).
It will be a number between 0 and 1.

So back to the example. We are having 200 steps.
horizontal = 200 * (300 / 400) = 150
vertical = 200 * (1 - 300 / 400) = 50

I think this is what we want.

**Comment by brcolow** [comment]
Ah yes, you are very right. I will make the necessary changes in #510. Much appreciated.

**Comment by brcolow** [comment]
Does this look correct?

```java
double horizontalDistance = sourcePoint.distance(targetPoint.getX(), sourcePoint.getY());
double verticalDistance = sourcePoint.distance(sourcePoint.getX(), targetPoint.getY());
double percentHorizontal = horizontalDistance / (horizontalDistance + verticalDistance);
```

**Comment by seggcsont** [comment]
Yes, it is.
I have refactored this code in our test code.
I want to share with you, but I am not able to create a branch in this repo.

How can I share it?

**Comment by brcolow** [comment]
No need, I have already created a pull request that has the changes: #510

**Comment by brcolow** [comment]
I will add a helped by tag to the commit now :) - forgot to do that.

**Comment by brcolow** [comment]
Closed via #511 - thanks @seggcsont.
